Mathlib/Order/Interval/Finset/Fin.lean

```diff
@@ -127,45 +128,777 @@ theorem Iio_eq_finset_subtype : Iio b = (Iio (b : ℕ)).fin n := by ext; simp
 
 @[simp]
-theorem map_valEmbedding_Ici : (Ici a).map Fin.valEmbedding = Ico ↑a n := by
+theorem map_valEmbedding_Ici : (Ici a).map Fin.valEmbedding = Ico (a : ℕ) n := by
   rw [← attachFin_Ico_eq_Ici, map_valEmbedding_attachFin]
 
 @[simp]
-theorem map_valEmbedding_Ioi : (Ioi a).map Fin.valEmbedding = Ioo ↑a n := by
+theorem map_valEmbedding_Ioi : (Ioi a).map Fin.valEmbedding = Ioo (a : ℕ) n := by
   rw [← attachFin_Ioo_eq_Ioi, map_valEmbedding_attachFin]
 
 @[simp]
-theorem map_valEmbedding_Iic : (Iic b).map Fin.valEmbedding = Iic ↑b := by
+theorem map_valEmbedding_Iic : (Iic a).map Fin.valEmbedding = Iic (a : ℕ) := by
   rw [← attachFin_Iic, map_valEmbedding_attachFin]
 
 @[simp]
-theorem map_valEmbedding_Iio : (Iio b).map Fin.valEmbedding = Iio ↑b := by
+theorem map_valEmbedding_Iio : (Iio a).map Fin.valEmbedding = Iio (a : ℕ) := by
   rw [← attachFin_Iio, map_valEmbedding_attachFin]
 
+end val
+
+section castLE
+
+/-!
+### Image under `Fin.castLE`
+-/
+
+@[simp]
+theorem finsetImage_castLE_Icc (h : n ≤ m) :
+    (Icc a b).image (castLE h) = Icc (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Ico (h : n ≤ m) :
+    (Ico a b).image (castLE h) = Ico (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Ioc (h : n ≤ m) :
+    (Ioc a b).image (castLE h) = Ioc (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Ioo (h : n ≤ m) :
+    (Ioo a b).image (castLE h) = Ioo (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_uIcc (h : n ≤ m) :
+    (uIcc a b).image (castLE h) = uIcc (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Iic (h : n ≤ m) :
+    (Iic a).image (castLE h) = Iic (castLE h a) := by simp [← coe_inj]
+
+@[simp]
+theorem finsetImage_castLE_Iio (h : n ≤ m) :
+    (Iio a).image (castLE h) = Iio (castLE h a) := by simp [← coe_inj]
+
+/-!
+### `Finset.map` along `Fin.castLEEmb`
+-/
+
+@[simp]
+theorem map_castLEEmb_Icc (h : n ≤ m) :
+    (Icc a b).map (castLEEmb h) = Icc (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem map_castLEEmb_Ioc (h : n ≤ m) :
+    (Ioc a b).map (castLEEmb h) = Ioc (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem map_castLEEmb_Ico (h : n ≤ m) :
+    (Ico a b).map (castLEEmb h) = Ico (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem map_castLEEmb_Ioo (h : n ≤ m) :
+    (Ioo a b).map (castLEEmb h) = Ioo (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem map_castLEEmb_uIcc (h : n ≤ m) :
+    (uIcc a b).map (castLEEmb h) = uIcc (castLE h a) (castLE h b) := by simp [← coe_inj]
+
+@[simp]
+theorem map_castLEEmb_Iic (h : n ≤ m) :
+    (Iic a).map (castLEEmb h) = Iic (castLE h a) := by simp [← coe_inj]
+
+@[simp]
+theorem map_castLEEmb_Iio (h : n ≤ m) :
+    (Iio a).map (castLEEmb h) = Iio (castLE h a) := by simp [← coe_inj]
```

Comment:
Does that mean the `Ici` and `Ioi` variants are too ugly to be stated?

Reply:
Yes, the image can be `Ici` (if `n = m`) or `Ico` (if `n < m`).